### PR TITLE
[RATOM-226] add view as plaintext option in message detail view

### PR DESCRIPTION
**Pull request by michael-caktus**
Adds a checkbox and label to the MessageHeader in the MessageDetail view.

Defaults to off.

User selection is stored in local storage so that the choice will persist between messages and between sessions.

### CODE REVIEW

File: src/components/Containers/Message/ParsedMessageBody.js [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+import { MessageContext } from './MessageMain';
+
 const ParsedMessageBody = ({ message }) => {
+  const { viewAsPlaintext } = useContext(MessageContext);
   return (
     <MessageBodyWrapper>
       <MessageBodyStart>START MESSAGE BODY</MessageBodyStart>
-      <BodyContent dangerouslySetInnerHTML={{ __html: message.body }} />
+      {viewAsPlaintext ? (
+        <PlainBodyContent>{message.body}</PlainBodyContent>
```
Comment (JeremyGibson) [inline src/components/Containers/Message/ParsedMessageBody.js:14]: In other parts of the code aren't these built like this `<PlainBodyContentStyled>`. Is there are reason these don't follow that convention? Curious mostly.

File: src/components/Containers/Message/ParsedMessageBody.js [hunk 2]
```diff
@@ -33,8 +42,12 @@ const MessageBodyEnd = styled(MessageBodyDivider)`
   margin-top: 1rem;
 `;
 
-const BodyContent = styled.div`
+const HtmlBodyContent = styled.div`
   padding: 0 2rem;
 `;
 
+const PlainBodyContent = styled.div`
+  white-space: pre-wrap;
```
Comment (JeremyGibson) [inline src/components/Containers/Message/ParsedMessageBody.js:50]: nice!

**Review by JeremyGibson — APPROVED** [review]
My question isn't a blocker.

File: src/components/Containers/Message/ParsedMessageBody.js (hunk 1 above)
Comment (michael-caktus) [reply src/components/Containers/Message/ParsedMessageBody.js:14]: In truth, I haven't settled on a convention for nested styled-components. I think maybe what I like (but haven't done here) is something like this:
1. If the component is called Header, the outermost JSX tag should be called `<HeaderStyled>`.
2. Inner styled-components should be called `<HeaderNavList>` or `<HeaderListItem>`. Basically just namespace everythign inside the component.

That's something I've started trying mid-way through this project, so it hasn't really made its way in.

**Pull request merged**